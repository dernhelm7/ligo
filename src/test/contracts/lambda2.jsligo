type storage = unit;

/* Not supported yet:
   let main (p:unit) storage = (fun x -> ()) ()
   */

let main = ([_, _] : [unit, storage]) : unit =>
  ((f : ((a: unit) => unit)) => f (unit)) ((_useless : unit) => unit);
