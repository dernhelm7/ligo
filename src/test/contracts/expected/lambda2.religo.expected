type storage = unit;

let main = ((a, s): (unit, storage)): unit => 
  ((f: (unit => unit)) => f())((useless: unit) => unit);
